### DISCUSSION TITLE
PHP: Generator / Iterator support

### DISCUSSION THREAD

**Discussion by levu42** [spatie/ray#628] [Ideas]
I work a lot with generators and I often write code like

```php
foreach($generator as $item) {
    ray($item);
}
```

so I would suggest something like `ray()->iterateOver($generator);` with a signature of `iterateOver(Iterable $iterator, int $max=10)`.

What do you think about this feature, do you think it makes sense for me to submit a PR for it?

**Comment by freekmurze** [comment]
Seems interesting, go ahead!
